genie-web/src/main/java/com/netflix/genie/web/configs/aws/AwsS3Config.java

```diff
@@ -81,15 +83,19 @@ public STSAssumeRoleSessionCredentialsProvider awsCredentialsProvider(
     /**
      * A bean providing a client to work with S3.
      *
+     * @param noOfS3Retries No. of S3 request retries
      * @param awsCredentialsProvider A credentials provider used to instantiate the client.
      * @return An amazon s3 client object
      */
     @Bean
     @ConditionalOnBean(AWSCredentialsProvider.class)
     public AmazonS3Client genieS3Client(
+        @Value("${genie.retry.s3.noOfRetries:5}") final int noOfS3Retries,
```

Comment:
We should build up a type safe version of this so we don't bind to a string directly